### PR TITLE
Fix incorrect bundling

### PR DESCRIPTION
**Pull request by jtnord**
# Description

* Do not bundle `commons-codec` (again).
* Do not use maven apis that will blow up a scanner at some point when maven has an API.
* Do not bundle `guava`

This has has no testing performed (not even running unit tests - because I am on windows and building it is not a possibility in a reasonable time frame)

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@olamy

### CODE REVIEW

File: blueocean-bitbucket-pipeline/pom.xml [hunk 1]
```diff
@@ -43,18 +43,14 @@
...
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>apache-httpcomponents-client-4-api</artifactId>
     </dependency>
 
-    <dependency>
-      <groupId>org.apache.maven</groupId>
-      <artifactId>maven-artifact</artifactId>
-    </dependency>
-
```
Comment (jtnord) [inline blueocean-bitbucket-pipeline/pom.xml:57]: replaced by Jenkins' `version-lib`

File: blueocean-jira/pom.xml [hunk 2]
```diff
@@ -38,10 +38,6 @@
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>apache-httpcomponents-client-4-api</artifactId>
         </dependency>
-        <dependency>
-            <groupId>com.google.guava</groupId>
-            <artifactId>guava</artifactId>
-        </dependency>
```
Comment (jtnord) [inline blueocean-jira/pom.xml:44]: was never a good thing - so nuke this from orbit with extreme prejudice.

the tests where not updated - but leaving the dependency here would promote the dep to test scope which would likely break things like `hpi:run` as Jenkins would not have `guava` when it should have.

File: blueocean-jira/src/main/java/io/jenkins/blueocean/service/embedded/jira/BlueJiraIssue.java [hunk 3]
```diff
@@ -1,9 +1,5 @@
 package io.jenkins.blueocean.service.embedded.jira;
 
-import com.google.common.base.Objects;
-import com.google.common.base.Predicates;
-import com.google.common.collect.ImmutableList;
-import com.google.common.collect.Iterables;
```
Comment (jtnord) [inline blueocean-jira/src/main/java/io/jenkins/blueocean/service/embedded/jira/BlueJiraIssue.java:6]: java8 for the win!

File: blueocean-jira/src/main/java/io/jenkins/blueocean/service/embedded/jira/BlueJiraIssue.java [hunk 4]
```diff
@@ -97,9 +97,10 @@ public int hashCode() {
                 return null;
             }
             Collection<String> issueKeys = findIssueKeys(changeSetEntry.getMsg(), site.getIssuePattern());
-            Iterable<BlueIssue> transformed = Iterables.transform(issueKeys,
-                                                                  s -> BlueJiraIssue.create(site, action.getIssue(s)));
-            return ImmutableList.copyOf(Iterables.filter(transformed, Predicates.notNull()));
+            Collection<BlueIssue> blueIssues = issueKeys.stream().map(s -> BlueJiraIssue.create(site, action.getIssue(s)))
+                                                        .filter(Objects::nonNull)
+                                                        .collect(Collectors.toList());
```
Comment (jtnord) [inline blueocean-jira/src/main/java/io/jenkins/blueocean/service/embedded/jira/BlueJiraIssue.java:102]: should be about the same and the test passes

**Review by jtnord — COMMENTED** [review]
completely and utterly untested as building this is a complete 🔥 on windows - and even the WSL2 kernel is not happy with the output.

File: blueocean-bitbucket-pipeline/pom.xml [hunk 5]
```diff
@@ -43,18 +43,14 @@
     <dependency>
       <groupId>commons-codec</groupId>
       <artifactId>commons-codec</artifactId>
+      <scope>provided</scope>
```
Comment (jtnord) [inline blueocean-bitbucket-pipeline/pom.xml:46]: do not bundle this

**Review by MRamonLeon — APPROVED** [review]

File: blueocean-jira/src/main/java/io/jenkins/blueocean/service/embedded/jira/BlueJiraIssue.java (hunk 4 above)
Comment (MRamonLeon) [reply blueocean-jira/src/main/java/io/jenkins/blueocean/service/embedded/jira/BlueJiraIssue.java:102]: You can save one step, the `blueIssues` variable, doing it all together:
`.collect(collectingAndThen(toList(), ImmutableList::copyOf));`

**Review by bitwiseman — APPROVED** [review]

File: blueocean-bitbucket-pipeline/pom.xml (hunk 5 above)
Comment (olamy) [reply blueocean-bitbucket-pipeline/pom.xml:46]: maybe just remove the `<dependency>` as it's already declared in `<dependencies>` section in parent pom . a bit redundant?

**Review by olamy — APPROVED** [review]
just a nit picking comment.

File: blueocean-jira/src/main/java/io/jenkins/blueocean/service/embedded/jira/BlueJiraIssue.java (hunk 4 above)
Comment (jtnord) [reply blueocean-jira/src/main/java/io/jenkins/blueocean/service/embedded/jira/BlueJiraIssue.java:102]: `ImmutableList` is from Guava?

File: blueocean-bitbucket-pipeline/pom.xml (hunk 5 above)
Comment (jtnord) [reply blueocean-bitbucket-pipeline/pom.xml:46]: possibly but the other day you where saying if you use it declare it :)

this is known to work - I say ship it.

File: blueocean-jira/src/main/java/io/jenkins/blueocean/service/embedded/jira/BlueJiraIssue.java (hunk 4 above)
Comment (MRamonLeon) [reply blueocean-jira/src/main/java/io/jenkins/blueocean/service/embedded/jira/BlueJiraIssue.java:102]: Oh shit, I copied the guava version. I meant `.collect(collectingAndThen(Collectors.toList(), Collections::unmodifiableCollection));`

**Review by rsandell — APPROVED** [review]

**Pull request merged**